### ISSUE TITLE
Cross-reference link format problems

### ISSUE THREAD

**Issue by RexJaeschke** [spdx/spdx-spec#236]
The md contains a variety of approaches, of which at least the following will need to be changed:

- All occurrences of “section” in front of a clause-only (that is, not a subclause) number _must_ be changed to “Clause”.
- References to whole clauses, subclauses, or annexes, _must not_ include the respective clause, subclause, or annex title.
- All occurrences of “section” in front of a subclause number _must_ be removed.
- A subclause title that follows a cross-reference link rather than being inside the […] _should_ be removed.
- Add the prefix “see”, as necessary/appropriate, but putting it in the text preceding the link name, *not* as part of the link name.
- Put the cross-reference number inside parentheses, as appropriate (i.e., when it is not the direct or indirect object of text)
- In summary, review all cross-references to places within the spec for correctness and consistency.

For example:

- “See [section 5 Snippet Information](…) …” becomes “See [Clause 5](…) …”
- “See section [3.21 External Reference](…)” becomes “See [3.21](…)”
- “See [section 3.8](…) Files Analyzed …” becomes “See [3.8](…) …”
- “… Comments on License field [(section 4.7)](…).” becomes “… Comments on License field [(see 4.7)](…).”

**Comment by RexJaeschke** [comment]
As (top-level chapter) clause numbers will change with ISO, we are now using the following unique, single-letter codes for existing clauses *instead of numbers*:

* D - Document creation information
* P - Package information
* F - File information
* S - Snippet information
* L - Other licensing information
* R - Relationships between SPDX Elements
* A - Annotations
* I - Review information
* C - Composition of an SPDX document

So, a link to the "Document creation information" clause will say, "Clause D". And links to a subclause within that clause will say, "see D.x", where x is retained as is.

For links to Annexes, use, "Annex AAA" for *all* annexes, and we'll fix-up those later. (Of the course, the real link will still go to the correct  md file and header.) Right now, annexes do *not* have unique letter codes.

**Comment by RexJaeschke** [comment]
Fixed for all Clauses and Annexes. However, I found the following inconsistencies in Annexes, which need attention:

1. Appendix II: Level-2 headers have a hard-coded decimal number (1) prefix rather than using the Appendix I model of using the Roman number prefix. These headers have no anchor tags.

1. Appendix III: Level-2 headers have no number prefixes or anchors.

1. Appendix IV: Level-2 headers have no numbers, and the first one has no anchor. The level-3 headers have hard-coded number prefixes.

1. Appendix V: Level-2 headers have no number prefixes or anchors.

1. Appendix VI: Level-2 headers have no number prefixes or anchors. Level-3 have no prefixes (but do have anchors).

1. Appendix VIII: Level-2 headers have a combination of hard-coded and no number prefixes, and no anchors.

1. Appendix IX: Level-2 headers have no number prefixes or anchors.

**Comment by RexJaeschke** [comment]
I spun off Annex-specific new issues #368  and #369.
